service/worker/service.go

```diff
@@ -69,17 +81,47 @@ func (c *Service) Start() error {
 
 	// Init the Registry
 	// TODO: move to component init ?
-	c.registry, err = registry.NewRegistry(c.ctx, c.registryConfig)
+	reg, err := registry.NewRegistry(c.ctx, c.registryConfig)
+	if err != nil {
+		return err
+	}
+	c.registry = reg
+
+	brokerStore, berr := brokerstore.NewBrokerStore(c.workerConfig.Broker.Variant, &c.workerConfig.Broker.BrokerConfig)
+	if berr != nil {
+		return berr
+	}
+
+	projectCore := project.NewCore(project.NewRepo(reg))
+
+	topicCore := topic.NewCore(topic.NewRepo(reg), projectCore, brokerStore)
+
+	subscriptionCore := subscription.NewCore(subscription.NewRepo(reg), projectCore, topicCore)
+
+	// Register Node with registry
+	nodeRepo := node.NewRepo(c.registry)
+	nodeCore := node.NewCore(nodeRepo)
+	err = nodeCore.CreateNode(c.ctx, c.node)
+	if err != nil {
+		return err
+	}
+
+	// create nodeCache
+	c.nodeCache, err = nodeCore.ListKeys(c.ctx)
+	if err != nil {
+		return err
+	}
+
+	c.subCache, err = subscriptionCore.ListKeys(c.ctx)
```

Comment:
should this happen on all nodes or only on the one which is the leader?

Reply:
right, will move to leader func before setting up watch